app/src/main/java/com/droidhats/campuscompass/views/CalendarFragment.kt

```diff
@@ -0,0 +1,101 @@
+package com.droidhats.campuscompass.views
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.droidhats.campuscompass.viewmodels.CalendarViewModel
+import com.droidhats.campuscompass.R
+import com.droidhats.campuscompass.adapters.CalendarAdapter
+import com.droidhats.campuscompass.models.Calendar
+import com.droidhats.campuscompass.models.CalendarEvent
+
+class CalendarFragment : Fragment() {
+
+    private lateinit var calendarViewModel: CalendarViewModel
+
+    private var columnCount = 1
+
+    companion object {
+        private const val READ_CALENDAR_PERMISSION_REQUEST_CODE = 1
+        // This callback could also be private and be set on the host(main) Activity
+        var onCalendarEventClickListener: OnCalendarEventClickListener? = null
```

Comment:
I think we should keep it in this file.
The onCalendarEventClickListener is specific to the CalendarFragment also having this and future EventClickListeners in the main overtime would be messy.